### ISSUE TITLE
Unable to Build Keycloak from source on Windows

### ISSUE THREAD

**Issue by Ines-Belhaj** [keycloak/keycloak#19390]
### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the latest release

### Area

docs

### Describe the bug

I am currently having trouble building Keycloak from source on Windows and would greatly appreciate your assistance.

Here are the details of my environment:

Operating system: Windows 10 
Java version: JDK 17.0.2
Maven version: 3.9.1

### Version

21.0.1+

### Expected behavior

Keycloak should be successfully built.

### Actual behavior

The build fails with the following error message: 

Command execution failed.: Cannot run program "C:\Users\inesb\keycloak\operator\patch-sources.sh" (in directory "C:\Users\inesb\keycloak\operator"): CreateProcess error=193, %1 is not a valid Win32 application -> [Help 1]

### How to Reproduce?

Here are the steps I have taken so far:

1. Cloned the Keycloak source code from GitHub: https://github.com/keycloak/keycloak.git
2. Installed all the necessary dependencies and tools as per the README.md file
3. Executed the following command to build Keycloak: mvn clean install -Pdistribution

### Anything else?

The "patch-sources.sh" script is a Unix shell script and is not compatible with the Windows operating system. Acknowledging the error means failing to run a shell script on a Windows system, I tried to resolve it by trying other alternatives. But I still get building failures. 
I have searched for solutions online but have not been successful in finding a fix.

**Comment by mposolda** [comment]
Re-assigning to "team/cloud-native" as the issue seems to be in building operator

**Comment by Ines-Belhaj** [comment]
Indeed, I have detected that the issue is located in the operator folder, as Maven is executing a shell script not supported by the Windows system. I have written a batch script and redirected the Maven exec plugin to the script in question. However, the file paths passed as arguments in the addAnnotation function are not found, which leads to a building failure.  I am also getting the following warning "[WARNING] Parameter 'localRepository' is deprecated core expression; Avoid use of ArtifactRepository type. If you need access to local repository, switch to '${repositorySystemSession}' expression and get LRM from it instead."

**Comment by Pepo48** [comment]
jFTR, I think that this got resolved by https://github.com/keycloak/keycloak/pull/20866
